### ISSUE TITLE
[Fabric 1.19.2 Multiplayer] Altas only updates reliably if a copy of the source map is in the inventory

### ISSUE THREAD

**Issue by AkraticCritic** [Pepperoni-Jabroni/MapAtlases#64]
Using the latest version (2.1.0) of the mod on my fabric 1.19.2 server and atlases created from a source map only reliably update the player's position if you have a copy of the source map in your inventory. Can't reproduce in singleplayer and rolling back to the 2.0.1 version fixes the issue. Have tried disabling all feasible mods on client and server side (including Nullscape, Lithium and Krypton which seemed most likely to be a problem) and this doesn't fix it.

Client modlist: https://pastebin.com/Sabxam68
Server modlist: https://pastebin.com/5ff7nG9u

**Comment by Pepperoni-Jabroni** [comment]
Interesting, thanks for reporting! When you say its not updating, which of these do you mean:
1) there's no active map state selected/displayed (i.e. no mini map)
2) the player marker is not moving when you move
3) the map colors are not updating as you explore.

**Comment by Pepperoni-Jabroni** [comment]
Ah wait I see that you mentioned "the player's position".

**Comment by Pepperoni-Jabroni** [comment]
This doesn't repro on server with
	minecraft_version=1.19
	yarn_mappings=1.19+build.1
	loader_version=0.14.6
	fabric_version=0.58.0+1.19
Will keep digging

**Comment by AkraticCritic** [comment]
Yeah it's the white marker showing the player's location and direction that either doesn't display or freezes in place. The mini map displays and updates to the correct map when moving out of the currently displayed one.  Newly revealed areas do show up while exploring.

**Comment by Pepperoni-Jabroni** [comment]
Interesting... this still doesn't repro even after I updated to 1.19.2 w/
	minecraft_version=1.19.2
	yarn_mappings=1.19.2+build.8
	loader_version=0.14.9
	fabric_version=0.60.0+1.19.2
	cloth_config_version=8.0.75
	mod_menu_version=4.0.6
	trinkets_version=3.4.0

This may be an undiscovered incompatibility.

Or... it looks like your server may be missing Cloth Config? I see it installed on the client, but the server should require it too.

If you're willing, would you try this out with just Fabric API, Cloth Config, and Map Atlases?

**Comment by Pepperoni-Jabroni** [comment]
And if you have any server/client logs, that might help too. Thanks!

**Comment by AkraticCritic** [comment]
Adding cloth config didn't fix the issues in the existing world. Tried a new world with just fabric api, cloth config and map atlases loaded and couldn't reproduce the issue so I guess that indicates a mod incompatibility. Will do some more testing tomorrow to see if I can figure out which mod is responsible and get some logs if I can't.

**Comment by AkraticCritic** [comment]
Ok after some more testing I can report the issue:

- doesn't affect a new world with the same modlist
- doesn't affect a new world with map atlases mod added after the map has been made
- only affects maps on my world that have been enlarged twice or more
- isn't fixed by deleting idcounts.dat and all map_*.dat files in the world/data folder (new large maps still have the problem)

I should also mention I added map atlases to this world after several days of playing.
So in summary the issue may or may not be related to a mod incompatibility but only seems to show up after a while and may be due to the mod being added after world creation. I'll leave it to you to decide if this is worth investigating further haha.

Logs:
Server - https://pastebin.com/rNC8U7nt
Client - https://pastebin.com/iwRiP6wA

**Comment by Pepperoni-Jabroni** [comment]
Thanks very much for doing a detailed investigation! I'll continue to try and repro this issue myself.
